src/lein_monolith/task/with_dependency_set.clj

```diff
@@ -0,0 +1,55 @@
+(ns lein-monolith.task.with-dependency-set
+  (:require
+    [clojure.java.io :as io]
+    [lein-monolith.plugin :as plugin]
+    [lein-monolith.task.util :as u]
+    [leiningen.core.main :as lein]
+    [leiningen.core.project :as project]))
+
+
+(defn- maybe-remove-dependencies
+  [project opts]
+  (if (:only opts)
+    (assoc project :dependencies '())
+    project))
+
+
+(defn- reload-project
+  "Reloads the raw project.clj file for the given project.
+   This has the effect of removing any middleware that was added to the project
+   as well as any changes to the project map that were made in the current
+   session."
+  [project]
+  (-> project
+      (:root)
+      (io/file "project.clj")
+      (str)
+      (project/read-raw)))
```

Comment:
I don't think interacting with raw project maps should be necessary for this; could you explain why we need to do so?

Also, stylistically, I think this is an example of a threading antipattern - thread-first and thread-last are best used to chain together successive transformations on a similar argument type. For example, using thread-first to combine some calls to `assoc`, `dissoc`, `update-in`, and so on to modify an associative data structure, or using thread-last to combine `map`, `filter`, `partition`, etc on a sequence. Here the value being acted on is extremely different at each step - first a project map, then a path string, then a File object, back to a path string, then back to a project map. I think it'd be more readable (and actually fewer lines) to use a let binding to break this up:
```clojure
(let [project-file (io/file (:root project) "project.clj")]
  (project/read-raw (str project-file)))
```

Reply:
Yeah, good question. Because we have implicitly evaluated middleware, the project structure gets evaluated with the middleware activated and making a modification to the project dependencies or dependency set fails to actually take effect. For the root project, this is fine as we're simply replacing the existing `:managed-dependencies`. However, when working on a subproject, the gymnastics required would be de-activating the middleware, undoing the changes to the profiles, and some other things I'm forgetting. This was a simpler solution but I agree it's hacky.

Reply:
Is there a reason we can't take the approach in the `with-all` task and create and merge in a new profile which overrides the settings we care about?